model/textparse/openmetricsparse.go

```diff
@@ -222,9 +225,108 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 // CreatedTimestamp returns nil as it's not implemented yet.
 // TODO(bwplotka): https://github.com/prometheus/prometheus/issues/12980
 func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
+	var lbs labels.Labels
+	p.Metric(&lbs)
+	lbs = lbs.DropMetricName()
+
+	newParser := deepCopyParser(p)
+loop:
+	for {
+		switch t, _ := newParser.Next(); t {
+		case EntrySeries:
+			// TODO: potentially broken? Missing type?
+			if newParser.mName != p.mName {
+				return nil
+			}
+
+			var newLbs labels.Labels
+			newParser.Metric(&newLbs)
+			name := newLbs.Get(model.MetricNameLabel)
+			if !strings.HasSuffix(name, "_created") {
+				continue
+			}
+
+			// edge case: if gauge_created of unknown type -> skip parsing
+			newLbs = newLbs.DropMetricName()
+			switch p.mtype {
+			case model.MetricTypeCounter:
+				if !labels.Equal(lbs, newLbs) {
+					return nil
+				}
+			case model.MetricTypeSummary:
+				labelDiffs := lbs.MatchLabels(false, newLbs.ExtractNames()...)
+				if labelDiffs.Len() != 0 {
+					if !labelDiffs.Contains("quantile") || labelDiffs.Len() != 1 {
+						return nil
+					}
+				}
+			case model.MetricTypeHistogram:
+				labelDiffs := lbs.MatchLabels(false, newLbs.ExtractNames()...)
+				if labelDiffs.Len() != 0 {
+					if !labelDiffs.Contains("le") || labelDiffs.Len() != 1 {
+						return nil
+					}
+				}
+			default:
+				// if mtype is not counter, summary or histogram, it won't have a valid _created line
+				return nil
+			}
+
+			ct := int64(newParser.val)
+			return &ct
+		default:
+			break loop
+		}
+	}
 	return nil
 }
 
+// We need this because we want offsets of the original parser unchanged when
+// we're working with a new parser in CreatedTimeStamp().
```

Comment:
```suggestion
// deepCopyParser creates a copy of a parser without re-using the slices' original memory addresses.
// The function `CreatedTimestamp()` uses a copy of the parser to "peek" at _created lines that might be several lines ahead, without changing the state of the original parser.
//
// Additionally, deepCopyParser switches `skipCT` from false to true, because this new parser needs to return _created lines.
```